x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java

```diff
@@ -3850,6 +3856,56 @@ public void testRoleRestrictionAccessDenial() {
         assertThat(securityException.getRootCause(), throwableWithMessage(containsString("access restricted by workflow")));
     }
 
+    public void testSetExceptionOnMissingIndexWhenIgnoreUnavailable() {
+        final ProjectRoutingInfo originProject = createRandomProjectWithAlias(randomAlphaOfLengthBetween(6, 10));
+
+        doAnswer(invocation -> {
+            @SuppressWarnings("unchecked")
+            ActionListener<TargetProjects> callback = (ActionListener<TargetProjects>) invocation.getArguments()[0];
+            callback.onResponse(new TargetProjects(originProject, Collections.emptyList()));
+            return null;
+        }).when(authorizedProjectsResolver).resolveAuthorizedProjects(anyActionListener());
+
+        when(crossProjectModeDecider.crossProjectEnabled()).thenReturn(true);
+        when(crossProjectModeDecider.resolvesCrossProject(any())).thenReturn(true);
+
+        mockMetadataWithIndex("available-index");
```

Comment:
The deciding factor here is that the user has no access to the index regardless whether it exists or not. So I'd suggest:
1. rename `available-index` and `not-available-index` to `accessible-index` and `not-accessible-index`.
2. Randomly also add `not-accessible-index` to the project metadata. The behaviour should not change whether this index exists or not
3. random between indices request between `accessible-index,not-accessible-index` and just `not-accessible-index`. The behaviour should be the same.